issue18 - adding spring boot template support

bfitzpat opened the pull request:
-- added spring-boot as supported DSL
-- added new spring-boot template
-- added new basic (not wsdl2rest) spring-boot test
-- added support for spring-boot with wsdl2rest
-- updated test for utils to include spring-boot
-- added new test for spring-boot with wsdl2rest
-- updated version to 0.2.0
-- updated README.md

Signed-off-by: Brian Fitzpatrick <bfitzpat@redhat.com>

lhein commented on app/templates/spring-boot/pom.xml.wsdl2rest:
```diff
@@ -0,0 +1,276 @@
+		</dependency>
+
+		<!-- Testing & Camel Plugin -->
+    	<dependency>
```
indentation problem

lhein commented on app/templates/spring-boot/pom.xml.wsdl2rest:
```diff
@@ -0,0 +1,276 @@
+	</build>
+
+	<profiles>
+		<profile>
```
indentation issue

lhein commented on app/templates/spring-boot/pom.xml.wsdl2rest:
```diff
@@ -0,0 +1,276 @@
+		</activation>
+		<build>
+			<plugins>
+			<plugin>
```
indentation issue

lhein commented on app/templates/spring-boot/pom.xml.wsdl2rest:
```diff
@@ -0,0 +1,276 @@
+			<plugin>
+				<artifactId>maven-surefire-plugin</artifactId>
+				<configuration>
+				<argLine>--add-modules java.xml.bind --add-opens java.base/java.lang=ALL-UNNAMED</argLine>
```
indentation issue

lhein commented on app/templates/spring-boot/pom.xml:
```diff
@@ -0,0 +1,261 @@
+      		<groupId>org.springframework.boot</groupId>
+      		<artifactId>spring-boot-starter-test</artifactId>
+      		<scope>test</scope>
+		</dependency>
```
indentation

lhein commented on app/templates/spring-boot/pom.xml:
```diff
@@ -0,0 +1,261 @@
+      		<artifactId>spring-boot-starter-test</artifactId>
+      		<scope>test</scope>
+		</dependency>
+		<dependency>
```
whole block indent

lhein commented on app/templates/spring-boot/pom.xml:
```diff
@@ -0,0 +1,261 @@
+			<plugin>
+				<artifactId>maven-surefire-plugin</artifactId>
+				<configuration>
+				<argLine>--add-modules java.xml.bind --add-opens java.base/java.lang=ALL-UNNAMED</argLine>
```
indent

bfitzpat commented:
@lhein I think I've fixed all the weird indent issues in the two spring boot poms. Had to show the white space in the editor to find the problems, but there were a few spots with a mix of spaces and tabs. It's all tabs now.

apupier commented on app/templates/spring-boot/README.md:
```diff
@@ -0,0 +1,19 @@
+<%= userProps.appName %>
```
in the generated readme I have a blank first line, seems the value is not replaced (or we miss a parameter to configure it?)

![image](https://user-images.githubusercontent.com/1105127/50440648-7981ba00-08f7-11e9-8013-eba23aa3fc28.png)

bfitzpat replied:
Interesting. I'll see if I can figure out what's going on there.

bfitzpat replied:
Figured out this issue. Should be userProps.name instead of userProps.appName. Will also address this in the other readme files.

bfitzpat commented on app/templates/spring-boot/src/main/java/routes/SampleCamelApplication.java:
```diff
@@ -0,0 +1,40 @@
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ImportResource;
+
+//CHECKSTYLE:OFF
```
No idea. Likely a copy/paste error when I pulled that new code in. I will remove it to see what shakes out.

bfitzpat replied:
Will remove checkstyle annotations

bfitzpat commented on app/templates/spring-boot/pom.xml:
```diff
@@ -0,0 +1,261 @@
+		<repository>
+			<id>fuse-public-repository</id>
+			<name>FuseSource Community Release Repository</name>
+			<url>https://repo.fusesource.com/nexus/content/groups/public</url>
```
Not sure it is. Grabbed the pom from another example - will remove and see if we still build/run ok

bfitzpat replied:
Tested locally and seems to be unnecessary. Will remove.

apupier replied:
you plan to remove it in this PR, right?

bfitzpat replied:
Oops. I removed the plugin repo, but not the main repo - testing with that now. :)

bfitzpat replied:
Seems to work! Will get that removed.

bfitzpat commented on app/templates/spring-boot/src/main/resources/camel-context.xml:
```diff
@@ -0,0 +1,27 @@
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="        http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd        http://camel.apache.org/schema/spring http://camel.apache.org/schema/spring/camel-spring.xsd">
+    <camelContext id="_camelContext1" xmlns="http://camel.apache.org/schema/spring">
+        <route id="_route1">
+            <from uri="timer://simpleTimer?period=1000"/>
```
The same issue exists in the Spring and Blueprint XML templates, so I have spun off a new issue to address this after the spring boot example is merged.

bfitzpat commented on app/templates/spring-boot/src/main/resources/camel-context.xml:
```diff
@@ -0,0 +1,27 @@
+            <setBody>
+                <simple>Hello from timer at ${header.firedTime}></simple>
+            </setBody>
+            <to uri="stream:out"/>
```
The same issue exists in the Spring and Blueprint XML templates, so I have spun off a new issue to address this after the spring boot example is merged. See https://github.com/camel-tooling/generator-camel-project/issues/39

bfitzpat commented:
@apupier I have addressed all issues you mentioned in your review except for  https://github.com/camel-tooling/generator-camel-project/issues/39, which I would like to handle separately because it affects all XML-based camel configuration file templates.

bfitzpat commented:
@apupier removed last vestiges of the fusesource repo
